internal/oid/oid.go

```diff
@@ -0,0 +1,71 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package oid collects object identifiers for crypto algorithms.
+package oid
+
+import "encoding/asn1"
+
+// OIDs for hash algorithms
+var (
+	// SHA256 (id-sha256) is defined in RFC 8017 B.1 Hash Functions
+	SHA256 = asn1.ObjectIdentifier{2, 16, 840, 1, 101, 3, 4, 2, 1}
+
+	// SHA384 (id-sha384) is defined in RFC 8017 B.1 Hash Functions
+	SHA384 = asn1.ObjectIdentifier{2, 16, 840, 1, 101, 3, 4, 2, 2}
+
+	// SHA512 (id-sha512) is defined in RFC 8017 B.1 Hash Functions
+	SHA512 = asn1.ObjectIdentifier{2, 16, 840, 1, 101, 3, 4, 2, 3}
+)
+
+// OIDs for signature algorithms
+var (
+	// RSA is defined in RFC 8017 C ASN.1 Module
+	RSA = asn1.ObjectIdentifier{1, 2, 840, 113549, 1, 1, 1}
```

Comment:
Although it's allowed by CMS, notation has its own [signature-algorithm-requirements](https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#signature-algorithm-requirements). I think we should still follow this requirement, i.e., only accept signature algorithms associated with a hash func.
```suggestion
```